runtime/vm/ClassInitialization.cpp

```diff
@@ -386,6 +386,21 @@ classInitStateMachine(J9VMThread *currentThread, J9Class *clazz, J9ClassInitStat
 				if (NULL == initializationLock) {
 					goto done;
 				}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+				PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, initializationLock);
+				/* Preparation is the earliest point where the defaultValue would needed. 
+				 * I.e pre-filling static fields. Therefore, the defaultValue must be allocated at 
+				 * the end of verification 
+				 */
+				j9object_t defaultValue = vm->memoryManagerFunctions->J9AllocateObject(currentThread, clazz, J9_GC_ALLOCATE_OBJECT_TENURED | J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+				initializationLock = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
+				clazz = VM_VMHelpers::currentClass(clazz);
+				if (NULL == defaultValue) {
+					setHeapOutOfMemoryError(currentThread);
+					goto done;
+				}
+				clazz->flattenedClassCache[0].clazz = (J9Class *)defaultValue;
```

Comment:
Very confusing indeed. Please never do it again. If you need to reuse slots in `j9class` structure please use explicit cast to proper structure to map it properly